### PR TITLE
Ruckus Smartzone WLC and Access Point Driver

### PR DESCRIPTION
**Pull request by pato23arg**
This PR adds nornir-nautobot API support for Ruckus Smartzone OS devices:

- Provides async REST API capabilities.
- Provides a list of default endpoints for wireless controller and access point roles. Can be modified/monkey patched at runtime.
- Supports simple and nested endpoints (one-level depth).
- Retrieves API endpoint data in JSON format.

Wireless controller platform is supported in nornir-nautobot as standalone package and within nautobot deployment. Access point platform is only supported within nautobot deployments (device object mandatory).

### CODE REVIEW

**Review by jvanderaa — APPROVED** [review]

File: nornir_nautobot/plugins/tasks/dispatcher/ruckus_smartzone_api.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        return url_dict
+
+    @staticmethod
+    async def _async_get_data(url_dict: dict) -> dict:
+        # login use session params
+        headers = {"Accept": "application/json", "Content-Type": "application/json;charset=UTF-8"}
+
+        async with httpx.AsyncClient(verify=False) as client:
+            client.headers = headers
+            coroutines = [client.get(url) for url in url_dict.values()]
```
[the remark this answers is not in the document]
Comment (pato23arg) [reply nornir_nautobot/plugins/tasks/dispatcher/ruckus_smartzone_api.py:122]: @jvanderaa I tested and validated ruckus documentation and they don't seem to implement rate limiting (as Big-IP for instance): 
[https://docs.ruckuswireless.com/vscg-enterprise/vsz-e-public-api-reference-guide-3-2-1.html](url)

**Comment by jvanderaa** [comment]
@itdependsnetworks are you OK with this?? 

My only concern is that we are adding the Python httpx library to this for the driver. I don't generally mind using libraries that we need to use. But with httpx and requests playing similar roles, it's yet one more library to have. And for the individuals that are using this library for their inventory, now they are getting httpx that is for unknown reason.

**Comment by jvanderaa** [comment]
@pato23arg let's get the conflict resolved so we can merge this.

**Comment by wirestar** [comment]
@pato23arg how do I get ahold of you about this integration?

**Pull request merged**